### PR TITLE
feat: add event filtering plugin

### PR DESCRIPTION
**Pull request by vgupta98**
## Description
This PR implements event filtering functionality by adding a new `EventFilteringPlugin` that filters track events based on whitelist/blacklist configurations from destination settings. The plugin is automatically applied to all integration destinations and provides fine-grained control over which events are sent to specific destinations based on RudderStack dashboard configuration.

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code refactor/optimization
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Implementation Details
- Added `EventFilteringPlugin.swift` - A new preProcess plugin that filters track events based on destination configuration
- Created `EventProcessing` directory structure under `DeviceMode/EventProcessing/` for organization
- Enhanced `IntegrationsController.swift` to automatically apply `EventFilteringPlugin` to all integration destinations
- Implemented comprehensive test coverage in `EventFilteringPluginTests.swift` with 20+ test scenarios
- Updated Xcode project file to include new source files and test files
- Plugin supports two filtering modes:
  - **Whitelist**: Only events in the whitelist are allowed through
  - **Blacklist**: All events except those in the blacklist are allowed through
- Automatic configuration updates via source configuration listener
- Robust error handling for malformed configurations

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
1. **Event Filtering with Whitelist**:
   - Configure a destination with `eventFilteringOption: "whitelistedEvents"` and a list of allowed events
   - Send various track events through the analytics pipeline
   - Verify only whitelisted events reach the destination

2. **Event Filtering with Blacklist**:
   - Configure a destination with `eventFilteringOption: "blacklistedEvents"` and a list of blocked events
   - Send various track events through the analytics pipeline  
   - Verify blacklisted events are filtered out while others pass through

3. **Edge Cases**:
   - Test with empty event names, whitespace-only names
   - Test with malformed configuration (invalid arrays, missing eventName fields)
   - Verify plugin handles configuration gracefully without crashing

4. **Non-Track Events**:
   - Send identify, screen, group, alias events
   - Verify these events always pass through regardless of filtering configuration

5. **Configuration Updates**:
   - Update destination configuration via source config
   - Verify filtering rules update automatically without restart

6. **Run Test Suite**:
   ```bash
   swift test --filter EventFilteringPluginTests
   ```

## Breaking Changes
None. This is a purely additive feature that automatically applies to integration destinations without affecting existing analytics functionality or APIs.

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Screenshots (if applicable)
N/A - This is a backend SDK feature with no UI components.

## Additional Context
This implementation is part of SDK-2375 (Event Filtering Plugin) and builds upon the integration management foundation established in SDK-4044. The feature provides:

- **Granular Event Control**: Destination-level filtering based on dashboard configuration
- **Automatic Application**: Plugin is automatically added to all integration destinations
- **Real-time Configuration**: Updates filter rules when source configuration changes
- **Robust Error Handling**: Gracefully handles malformed configurations and edge cases
- **Performance Optimized**: Minimal overhead with efficient event filtering logic

Key architectural decisions:
- Uses `preProcess` plugin type to filter events before they reach destination plugins
- Listens to source configuration changes for automatic rule updates
- Only affects track events, allowing lifecycle events (identify, screen, etc.) to pass through
- Maintains thread safety with `@Synchronized` property wrappers
- Provides comprehensive test coverage with 20+ test scenarios covering normal usage, edge cases, and error conditions

## Summary by CodeRabbit

* **New Features**
  * Event filtering added: runtime filtering per-destination with whitelist and blacklist modes; filtering is now applied by default during plugin setup.

* **Tests**
  * Added comprehensive tests covering lifecycle, interception behavior, configuration updates, edge cases, and deinitialization.

### CODE REVIEW

**Review by SKannaniOS — APPROVED** [review]

**Review by SKannaniOS — APPROVED** [review]
Looks good to me.

**Review by SKannaniOS — APPROVED** [review]

File: Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/EventFilteringPlugin.swift [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        self.analytics = analytics
+        setupConfigurationListener()
+    }
+    
+    func intercept(event: any Event) -> (any Event)? {
+        guard let trackEvent = event as? TrackEvent else {
+            return event
+        }
+        
+        let eventName = trackEvent.event.trimmingCharacters(in: .whitespacesAndNewlines)
```
[the remark this answers is not in the document]
Comment (vgupta98) [reply Sources/RudderStackAnalytics/Plugins/DeviceMode/EventProcessing/EventFilteringPlugin.swift:54]: The trimmed event is only created to check whether it needs to be filtered or not. We don't modify event using this plugin.
